python/cudf/cudf/tests/test_scalar.py

```diff
@@ -18,6 +20,13 @@
     TIMEDELTA_TYPES,
 )
 
+
+@pytest.fixture(autouse=True)
```

Comment:
Is this important to do as an `autouse`? Just wondering if we'd get meaningful reductions in test times if we actually cached our scalars. FWIW the rest of our tests that do raw scalars will now automatically be cached right (e.g. binops tests that do `x = 1`)?

Reply:
Yes -- because some of these tests make assertions about Scalar internals, e.g., testing `_is_host_value_current`/`_is_device_value_current`, which caching from previous tests can mess with.

We could of course use the fixture in just those tests. Happy to make that change but I figure it doesn't hurt to `autouse` here: makes writing future tests a tad less error prone.

Reply:
(just to explicitly address the point: test time was not a consideration at all in making this fixture `autouse`)

Reply:
That's fine, we could consider being more granular in the future but there's probably much higher-leverage ways to decrease our test time.